File: ruby/lib/coding_interview/application/repository/portfolio_repository.rb
module CodingInterview
  module Application
    module Repository
      # 最適ポートフォリオリポジトリ。
      class PortfolioRepository
        def get; raise NotImplementedError; end
        def update(_portfolio); raise NotImplementedError; end
      end
    end
  end
end

File: python/src/coding_interview/application/repository/portfolio_repository.py
from __future__ import annotations

from typing import Protocol

from coding_interview.domain.stock import Portfolio


class PortfolioRepository(Protocol):
    """最適ポートフォリオリポジトリ。"""

    def get(self) -> Portfolio: ...
    def update(self, portfolio: Portfolio) -> None: ...
